File: Notes/View/ConsoleUI.py
from FileWork.FileHandler import FileHandler
from Presenter.Presenter import Presenter
from View.MainMenu import MainMenu
import re


class ConsoleUI:
    INPUT_ERROR = "Вы ввели неверное значение"

    def __init__(self):
        self.presenter = Presenter(self, FileHandler())
        self.work = True
        self.menu = MainMenu(self)

    def createData(self):
        self.presenter.addItem("Заметка 1", "Текст 1")
        self.presenter.addItem("Заметка 2", "Текст 2")
        self.presenter.addItem("Заметка 3", "Текст 3")

    def printAnswer(self, text):
        print(text)

    def start(self):
        self.hello()
        while self.work:
            self.printMenu()
            self.execute()

    def finish(self):
        print("До свидания")
        self.work = False

    def sortByDate(self):
        print("Сортировка по дате")
        self.presenter.sortByDate()

    def sortByHeader(self):
        print("Сортировка по названию")
        self.presenter.sortByHeader()

    def getInfo(self):
        self.presenter.getInfo()

    def addItem(self):
        print("Добавление данных")
        header = input("Введите название: ")
        body = input("Введите текст заметки: ")
        self.presenter.addItem(header, body)

    def saveData(self):
        print("Сохранение")
        fileName = input("Введите имя файла: ")
        self.presenter.saveData(fileName)

    def loadData(self):
        print("Загрузка")
        fileName = input("Введите имя файла: ")
        self.presenter.loadData(fileName)

    def clearData(self):
        print("Удаление данных")
        question = input("Удалить данные Y/N?")
        if question == "Y":
            self.presenter.clearData()

    def editData(self):
        print("Редактирование записи")
        line = input("Введите номер записи: ")
        if self.checkTextForInt(line):
            numId = int(line)
            self.presenter.editData(numId)
        else:
            self.inputError()

    def editItem(self, note):
        print(note.getFullInfo())
        header = input("Введите название: ")
        body = input("Введите текст заметки: ")
        self.presenter.editItem(note, header, body)

    def removeItem(self):
        print("Удаление заметки")
        self.getInfo()
        line = input("Введите номер заметки: ")
        if self.checkTextForInt(line):
            numId = int(line)
            self.presenter.removeItem(numId)

    def hello(self):
        print("Доброго времени суток!")

    def execute(self):
        line = input()
        if self.checkTextForInt(line):
            numCommand = int(line)
            if self.checkCommand(numCommand):
                self.menu.execute(numCommand)

    def checkTextForInt(self, text):
        if re.match(r"[0-9]+", text):
            return True
        else:
            return False

    def checkCommand(self, numCommand):
        if numCommand <= self.menu.getSize():
            return True
        else:
            self.inputError()
            return False

    def printMenu(self):
        print(self.menu.menu())

    def inputError(self):
        print(self.INPUT_ERROR)

File: ToyShop/View/ConsoleUI.java
package View;

import Presenter.Presenter;
import FileWork.FileHandler;

import java.util.Random;
import java.util.Scanner;

public class ConsoleUI implements View {

    private static final String INPUT_ERROR = "Вы ввели неверное значение";
    private Scanner scanner;
    private Presenter presenter;
    private boolean work;
    private MainMenu menu;

    public ConsoleUI() {
        scanner = new Scanner(System.in);
        presenter = new Presenter(this, new FileHandler());
        work = true;
        menu = new MainMenu(this);
    }

    public void randomToys(int num) {
        Random rnd = new Random();

        String[] names = new String[] { "Кукла", "Медведь", "Машинка", "Робот", "Крокодил" };

        for (int i = 0; i < num; i++) {
            presenter.addItem(names[rnd.nextInt(names.length)], rnd.nextInt(5), rnd.nextInt(100));
        }
    }

    @Override
    public void printAnswer(String text) {
        System.out.println(text);
    }

    @Override
    public void start() {
        title();

        while (work) {
            try {
                printMenu();
                execute();
            } catch (NumberFormatException e) {
                inputError();
            } catch (Exception e) {
                System.out.println(e);
            }
        }
    }

    public void finish() {
        System.out.println("До свидания");
        work = false;
    }

    public void getToyShopInfo() {
        presenter.getToyShopInfo();
    }

    public void getLotteryInfo() {
        presenter.getLotteryInfo();
    }

    public void addItem() {
        System.out.println("Добавление данных");
        System.out.println("Введите название");
        String name = scanner.nextLine();

        System.out.println("Укажите количество");
        String countString = scanner.nextLine();
        Integer count = 0;
        if (countString != "")
            count = Integer.parseInt(countString);

        System.out.println("Укажите вес в лотерее");
        String weightString = scanner.nextLine();
        Integer weight = 0;
        if (weightString != "")
            weight = Integer.parseInt(weightString);

        presenter.addItem(name, count, weight);
    }

    public void lottery() {
        presenter.lottery();
    }

    public void getPrize() {
        presenter.getPrize();
    }

    public void changeWeight() {
        System.out.println("Изменение веса в лотерее");
        presenter.getToyShopInfo();

        System.out.println("Укажите id");
        String idString = scanner.nextLine();
        Integer id = 0;
        if (idString != "")
            id = Integer.parseInt(idString);

        if (presenter.findData(id)) {
            System.out.println("Укажите вес в лотерее");
            String weightString = scanner.nextLine();
            Integer weight = 0;
            if (weightString != "")
                weight = Integer.parseInt(weightString);
            presenter.changeWeight(id, weight);
        } else {
            printAnswer("Данные не найдены");
        }

    }

    private void title() {
        System.out.println("Розыгрыш игрушек");
    }

    private void execute() {
        String line = scanner.nextLine();
        if (checkTextForInt(line)) {
            int numCommand = Integer.parseInt(line);
            if (checkCommand(numCommand)) {
                menu.execute(numCommand);
            }
        }
    }

    private boolean checkTextForInt(String text) {
        if (text.matches("[0-9]+")) {
            return true;
        } else {
            inputError();
            return false;
        }
    }

    private boolean checkCommand(int numCommand) {
        if (numCommand <= menu.getSize()) {
            return true;
        } else {
            inputError();
            return false;
        }
    }

    private void printMenu() {
        System.out.println(menu.menu());
    }

    private void inputError() {
        System.out.println(INPUT_ERROR);
    }
}
